src/v/cluster_link/replication/partition_replicator.cc

```diff
@@ -177,6 +185,10 @@ ss::future<> partition_replicator::fetch_and_replicate() {
             if (data.batches.empty()) {
                 continue;
             }
+            if (_probe_data) {
+                _probe_data->total_fetched += fetch_counters::from_fetch_data(
```

Comment:
nit: do from_fetch_data{..) only once? Since it loops through batches for record count, pass the output to replicate_and_wait()